File: c/greedy/assign-cookies.c
int cmp(const void *a, const void *b)
{
    return *(int *)a - *(int *)b;
}

int findContentChildren(int* g, int gSize, int* s, int sSize) {
    int i = gSize - 1;
    int j = sSize - 1;
    int ret = 0;

    qsort(g, gSize, sizeof(int), cmp);
    qsort(s, sSize, sizeof(int), cmp);

    while(i >= 0 && j >= 0){
        if(s[j] >= g[i]){
            j--;
            ret++;
        }
        i--;
    }

    return ret;
}

File: python/greedy/assign-cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        i, j = len(g) - 1, len(s) - 1
        g, s = sorted(g), sorted(s)
        ret = 0

        while min(i, j) >= 0:
            if s[j] >= g[i]:
                ret += 1
                j -= 1

            i -= 1

        return ret
